pkg/restapi/operation/operations.go

```diff
@@ -457,3 +460,80 @@ func (vc *VaultCollection) queryVault(vaultID string, query *models.Query) ([]st
 
 	return store.Query(query)
 }
+
+func validateDataVaultConfiguration(dataVaultConfig *models.DataVaultConfiguration) error {
+	if err := checkConfigRequiredFields(dataVaultConfig); err != nil {
+		return err
+	}
+
+	if err := edvutils.CheckIfURI(dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidControllerString, err)
+	}
+
+	if err := checkFieldsWithURIArray(&dataVaultConfig.Invoker, dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidInvokerStringArray, err)
+	}
+
+	if err := checkFieldsWithURIArray(&dataVaultConfig.Delegator, dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidDelegatorStringArray, err)
+	}
```

Comment:
I don't think we should be modifying the actual data vault configuration being passed in. Rather, if/when we add code later that uses those Controller, Invoker and Delegator fields, we can have the logic there that determines which one to use based on which ones are empty. This way, the original data vault configuration is preserved. This way, we can differentiate between a data vault configuration that was purposefully initialized with all three fields being the same value (which would be redundant) and one where we just default to the controller field. While your approach may technically work the same way, I think it's safer to leave the config as is and defer the logic of what to do if one thing is empty until we actually need to check.

Reply:
@DRK3 

> I  don't think we should be modifying the actual data vault configuration being passed in

I think you meant "verifying" right?

> if/when we add code later that uses those Controller, Invoker and Delegator fields

:+1:

Reply:
@llorllale I did mean "modifying" - that method actually modifies the Data Vault Configuration.

Reply:
Oh. That's _very_ bad practice and unexpected. In general, refrain from mutating input args.